### ISSUE TITLE
49.10: UnitMarketOffer::writeToXML NPE

### ISSUE THREAD

**Issue by AfroSamurai34** [MegaMek/mekhq#3456]
### Environment
What version of MekHQ does your issue apply to? 49.10
What operating system are you using?  Windows 10
What java version are you using?  If unsure it will state your java version at the top of the mekhq.log file.
java 11
### Description
When attempting to save campaign file, recieve an error message stating cannot save.
![Screenshot (20)](https://user-images.githubusercontent.com/14045991/190042842-1ce0a85a-fd1c-4816-a75b-57f5705b37d7.png)
### Files

[mekhq.log](https://github.com/MegaMek/mekhq/files/9561995/mekhq.log)
[megamek.log](https://github.com/MegaMek/mekhq/files/9561993/megamek.log)
[Autosave-1-My Campaign-30610312.cpnx.gz](https://github.com/MegaMek/mekhq/files/9562014/Autosave-1-My.Campaign-30610312.cpnx.gz)

**Comment by Windchild292** [comment]
```
0:04:44,089 ERROR [mekhq.gui.CampaignGUI] {AWT-EventQueue-0}
mekhq.gui.CampaignGUI.saveCampaign(CampaignGUI.java:1305) - 
java.lang.NullPointerException
	at mekhq.campaign.market.unitMarket.UnitMarketOffer.writeToXML(UnitMarketOffer.java:125)
	at mekhq.campaign.market.unitMarket.AbstractUnitMarket.writeBodyToXML(AbstractUnitMarket.java:243)
	at mekhq.campaign.market.unitMarket.AbstractUnitMarket.writeToXML(AbstractUnitMarket.java:231)
	at mekhq.campaign.Campaign.writeToXML(Campaign.java:4184)
	at mekhq.gui.CampaignGUI.saveCampaign(CampaignGUI.java:1296)
	at mekhq.gui.CampaignGUI.saveCampaign(CampaignGUI.java:1269)
```

**Comment by Windchild292** [comment]
UnitMarketOffer::getUnit is returning null. That shouldn't be possible, which means it's a deeper issue than the cause above.

**Comment by pheonixstorm** [comment]
I had this happen in one of the 0.49.9 nightlies. I was hoping to see it a second time but it hasn't happened since. In my case the problem was with a conventional fighter. I could load the unit market and all check boxes worked except the conventional fight section, it caused an NPE when I tried to view it.

Until there is a fix you can advance to the new month if you are already half way into the month or have a save prior to the new month.

**Comment by Windchild292** [comment]
Do you have a working save? The above autosave was corrupted on upload.

**Comment by Windchild292** [comment]
*potentially on upload

**Comment by AfroSamurai34** [comment]
[My Campaign30610312.cpnx.gz](https://github.com/MegaMek/mekhq/files/9579594/My.Campaign30610312.cpnx.gz)
This one might be a little better.  Just tried it, deleted some salvage conventional fighters that were salvage remnants (which might not be related anyhow) and then went to unit market, and the UI crashed on me.

**Comment by AfroSamurai34** [comment]
phoenixstorm's workaround worked for me...I advanced to the end of the month, whereupon the unit market reset, and I both could view conventional fighters in the unitmarket and check/uncheck the conventional fighter box in the gui, and it not only DIDNT throw a NPE at me, but I was able to save again.  This had the additional benefit of some "ghost" ordanance clearing up in the repair bay, because once I was able to save again, I was able to reload....which cleared that problem out.  I did notice some odd behavior shortly after that though, apparently some "subcontracted" mechs/mechwarriors I had hired on that had been destroyed reappeared as available for hire.  And when you hire them, the mech is in the same shape it was (salvage/destroyed) as it was when they ran their last mission.

**Comment by AfroSamurai34** [comment]
Last session I played, I noticed that I got an NPE when firing up MM (Thru MHQ).  cancelled, reloaded in the lobby, and instead of configuring everything to run precision and caseless AC ammo as I usually do, I left everything on default ammo types.  No NPE, no locking up.  The conventional fighter that was in the unit market that was causing it to crash may or may not have had bombs attached to it....which was part of some of the error messages both in the unit market and the repair bay.  Last thing I noticed, is that some (most?) of the CV, VTOLS, and other vehicles that have autocannons don't have the option to load alternative munitions....maybe this is what's causing the conflicts?  Hope these observations help, love the game!
